test/lib/util/fast_json_encoder_test.exs

```diff
@@ -0,0 +1,61 @@
+# Copyright(c) 2015-2019 ACCESS CO., LTD. All rights reserved.
+
+defmodule Antikythera.FastJasonEncoderTest do
+  use Croma.TestCase
+  use ExUnitProperties
+
+  def validate(value) do
+    validate(value, value)
+  end
+  def validate(source, expected) do
+    {:ok, json} = FastJasonEncoder.encode(source)
+    assert json == Poison.encode!(source)
+    assert expected == Poison.decode!(json)
+  end
+
+  test "primitives" do
+    values = [0, 1, 1.5, true, false, nil]
+    Enum.each(values, &validate(&1))
+    validate(:ok, "ok") # Atom is encoded as a string
+  end
+
+  test "complext types" do
+    values = [[], [1.5], [1, "foo"], %{}, %{"key" => "value"}]
```

Comment:
When I added `%{"key" => "value", "key2" => "value2"}` (map with multiple elements) to this `values`, I encountered the below error.

```
  1) test complext types (Antikythera.FastJasonEncoderTest)
     test/lib/util/fast_json_encoder_test.exs:22
     Assertion with == failed
     code:  assert json == Poison.encode!(source)
     left:  "{\"key\":\"value\",\"key2\":\"value\"}"
     right: "{\"key2\":\"value\",\"key\":\"value\"}"
     stacktrace:
       test/lib/util/fast_json_encoder_test.exs:12: Antikythera.FastJasonEncoderTest.validate/2
       (elixir) lib/enum.ex:783: Enum."-each/2-lists^foreach/1-0-"/2
       (elixir) lib/enum.ex:783: Enum.each/2
       test/lib/util/fast_json_encoder_test.exs:24: (test)
```

Is this an expected behavior?